tensorstore/kvstore/s3/credentials/ec2_credential_provider.h

```diff
@@ -19,30 +19,25 @@
 #include <utility>
 
 #include "absl/base/thread_annotations.h"
-#include "absl/synchronization/mutex.h"
-#include "absl/time/time.h"
 #include "tensorstore/internal/http/http_transport.h"
-#include "tensorstore/kvstore/s3/aws_credential_provider.h"
+#include "tensorstore/kvstore/s3/credentials/aws_credential_provider.h"
+#include "tensorstore/kvstore/s3/credentials/expiry_credential_provider.h"
 #include "tensorstore/util/result.h"
 
 namespace tensorstore {
 namespace internal_kvstore_s3 {
 
 /// Provide S3 credentials from the EC2 Metadata server
-class EC2MetadataCredentialProvider : public AwsCredentialProvider {
+class EC2MetadataCredentialProvider : public ExpiryCredentialProvider {
  public:
   EC2MetadataCredentialProvider(
```

Comment:
Like FileCredentialProvider, I'd expect an accessor for 

Result<std::string> GetEC2MetadataServiceEndpoint()

And I'd expect to be able to pass an endpoint into the constructor.

Reply:
... this one is still here. For s3-compatible stores we need to be able to set a specific endpoint for the credentials request.

Reply:
Thanks for prodding here. The UI tends to collapse and hide older comments which sometimes results in me missing these comments or seeing them later.

Reply:
Added a `GetEndpoint()` accessor